src/extensions/sentry-integration.ts

```diff
@@ -54,7 +60,90 @@ interface SentryExceptionProperties {
+    organization?: string
+    projectId?: number
+    prefix?: string
+    /**
+     * By default, only errors are sent to PostHog. You can set this to '*' to send all events.
+     * Or to an error of SeverityLevel to only send events matching the provided levels.
+     * e.g. ['error', 'fatal'] to send only errors and fatals
+     * e.g. ['error'] to send only errors -- the default when omitted
+     * e.g. '*' to send all events
+     */
+    severityAllowList?: _SeverityLevel[] | '*'
+}
+
+const NAME = 'posthog-js'
+
+export function createEventProcessor(
+    _posthog: PostHog,
+    { organization, projectId, prefix, severityAllowList = ['error'] }: SentryIntegrationOptions = {}
+): (event: _SentryEvent) => _SentryEvent {
+    return (event) => {
+        const shouldProcessLevel =
+            severityAllowList === '*' || severityAllowList.includes(event.level as _SeverityLevel)
+        if (!shouldProcessLevel || !_posthog.__loaded) return event
+        if (!event.tags) event.tags = {}
+
+        const personUrl = _posthog.requestRouter.endpointFor(
+            'ui',
+            `/project/${_posthog.config.token}/person/${_posthog.get_distinct_id()}`
+        )
+        event.tags['PostHog Person URL'] = personUrl
+        if (_posthog.sessionRecordingStarted()) {
+            event.tags['PostHog Recording URL'] = _posthog.get_session_replay_url({ withTimestamp: true })
+        }
+
+        const exceptions = event.exception?.values || []
+
+        const data: SentryExceptionProperties & {
+            // two properties added to match any exception auto-capture
+            // added manually to avoid any dependency on the lazily loaded content
+            $exception_message: any
+            $exception_type: any
+            $exception_personURL: string
+            $level: _SeverityLevel
+        } = {
+            // PostHog Exception Properties,
+            $exception_message: exceptions[0]?.value || event.message,
+            $exception_type: exceptions[0]?.type,
+            $exception_personURL: personUrl,
+            // Sentry Exception Properties
+            $sentry_event_id: event.event_id,
+            $sentry_exception: event.exception,
+            $sentry_exception_message: exceptions[0]?.value || event.message,
+            $sentry_exception_type: exceptions[0]?.type,
+            $sentry_tags: event.tags,
+            $level: event.level,
+        }
+
+        if (organization && projectId) {
+            data['$sentry_url'] =
+                (prefix || 'https://sentry.io/organizations/') +
+                organization +
+                '/issues/?project=' +
+                projectId +
+                '&query=' +
+                event.event_id
+        }
+        _posthog.capture('$exception', data)
+        return event
+    }
+}
+
+// V8 integration - function based
+export function sentryIntegration(_posthog: PostHog, options?: SentryIntegrationOptions): _SentryIntegration {
```

Comment:
could this be `v8SentryIntegration`?

just to avoid `sentryIntegration` and `SentryIntegration` confusion fun?

Reply:
Yeah but then we're stuck with that forever. Personally I prefer this but I do see your point

Reply:
yeah, and we have to rename it when v9 comes around so 🤷